From b96659c24bfb713ca1948fb10486743328c729c6 Mon Sep 17 00:00:00 2001
From: Yiwei Zhang <zzyiwei@chromium.org>
Date: Mon, 22 Dec 2025 15:20:08 -0800
Subject: [PATCH 1/2] venus: track renderer driver version for driver
 workaround

Acked-by: Valentine Burley <valentine.burley@collabora.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/39069>
(cherry picked from commit 5106ab5c7adee0a5036414913739bb6f9231f1e6)
---
 src/virtio/vulkan/vn_physical_device.c | 6 ++++--
 src/virtio/vulkan/vn_physical_device.h | 1 +
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/virtio/vulkan/vn_physical_device.c b/src/virtio/vulkan/vn_physical_device.c
index a4729d8..dd36141 100644
--- a/src/virtio/vulkan/vn_physical_device.c
+++ b/src/virtio/vulkan/vn_physical_device.c
@@ -482,6 +482,10 @@ vn_physical_device_sanitize_properties(struct vn_physical_device *physical_dev)
    struct vn_instance *instance = physical_dev->instance;
    struct vk_properties *props = &physical_dev->base.vk.properties;
 
+   /* store renderer driver info for implementation specific workarounds */
+   physical_dev->renderer_driver_id = props->driverID;
+   physical_dev->renderer_driver_version = props->driverVersion;
+
    const uint32_t version_override = vk_get_version_override();
    if (version_override) {
       props->apiVersion = version_override;
@@ -530,8 +534,6 @@ vn_physical_device_sanitize_properties(struct vn_physical_device *physical_dev)
    }
    memcpy(props->deviceName, device_name, device_name_len + 1);
 
-   /* store renderer VkDriverId for implementation specific workarounds */
-   physical_dev->renderer_driver_id = props->driverID;
    props->driverID = VK_DRIVER_ID_MESA_VENUS;
 
    snprintf(props->driverName, sizeof(props->driverName), "venus");
diff --git a/src/virtio/vulkan/vn_physical_device.h b/src/virtio/vulkan/vn_physical_device.h
index d4e2543..b366a61 100644
--- a/src/virtio/vulkan/vn_physical_device.h
+++ b/src/virtio/vulkan/vn_physical_device.h
@@ -94,6 +94,7 @@ struct vn_physical_device {
    uint32_t wa_min_fb_align;
 
    VkDriverId renderer_driver_id;
+   uint32_t renderer_driver_version;
 
    /* Static storage so that host copy properties query can be done once. */
    VkImageLayout copy_src_layouts[64];
-- 
2.56.0

